moveit_core/planning_scene/src/planning_scene.cpp

```diff
@@ -1507,10 +1446,7 @@ bool PlanningScene::processAttachedCollisionObjectMsg(const moveit_msgs::msg::At
         const moveit::core::AttachedBody* ab = robot_state_->getAttachedBody(object.object.id);
 
         // Allow overriding the body's pose if provided, otherwise keep the old one
-        if (object.object.pose.position.x == 0 && object.object.pose.position.y == 0 &&
-            object.object.pose.position.z == 0 && object.object.pose.orientation.x == 0 &&
-            object.object.pose.orientation.y == 0 && object.object.pose.orientation.z == 0 &&
-            object.object.pose.orientation.w == 0)
+        if (moveit::core::isEmpty(object.object.pose))
```

Comment:
This check was added to check for poses which have not been initialized (all values are 0). With ROS 2, Quaternions are valid by default since the w component is initialized to 1. This check would not catch if we us an uninitialized pose. In this case it's probably fine, but we should keep an eye on these things in the future.